src/Functions/array/arrayIndex.h

```diff
@@ -574,94 +624,247 @@ struct ArrayIndexGenericNullImpl
     }
 };
 
-
-inline bool allowArrayIndex(const DataTypePtr & type0, const DataTypePtr & type1)
+/**
+ * Check types extracted from Nullable() and LowCardinality()
+ */
+inline bool allowArguments(const DataTypePtr & array_inner_type, const DataTypePtr & arg)
 {
-    DataTypePtr data_type0 = removeNullable(type0);
-    DataTypePtr data_type1 = removeNullable(type1);
-
-    return ((isNativeNumber(data_type0) || isEnum(data_type0)) && isNativeNumber(data_type1))
-        || data_type0->equals(*data_type1);
+    /**
+     * Possible cases for #arg and #array_inner_type:
+     * 1. T
+     * 2. LC(T)
+     * 3. N(T)
+     * 4. N(LC(T)) -- quite strange, rarely found but possible.
```

Comment:
This is not possible. `DataTypeLowCardinality::canBeInsideNullable()` returns `false` and this is checked in `DataTypeNullable::DataTypeNullable`.

Reply:
Thanks for the hint.